### PR TITLE
! fix casting items in collections

### PR DESCRIPTION
**Pull request by pape-io**

### CODE REVIEW

File: lib/lhs/collection.rb [hunk 1]
```diff
@@ -64,11 +59,27 @@ def initialize(raw, parent, record)
...
         if item.is_a? Hash
-          yield LHS::Data.new(item, @parent, @record)
+          yield cast_item(item)
         else
           yield item
         end
       end
     end
+
+    private
```
Comment (ryabrody) [inline lib/lhs/collection.rb]: check the `private`

Comment (acoiro) [inline lib/lhs/collection.rb]: duplication

File: lib/lhs/collection.rb [hunk 2]
```diff
@@ -64,11 +59,27 @@ def initialize(raw, parent, record)
...
+
+    def cast_item(item)
+      data = LHS::Data.new(item, @parent, @record)
+      return @record.new(data) if @record
```
Comment (acoiro) [inline lib/lhs/collection.rb]: You don't like that?
`@record ? @record.new(data) : data`

**Comment by acoiro** [comment]
Small comment, otherwise :+1:

File: lib/lhs/collection.rb (hunk 1 above)
Comment (pape-io) [inline lib/lhs/collection.rb]: It's an inlined, private class.... what ever, it looks weird, will try to refactor it

File: lib/lhs/concerns/collection/internal_collection.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require 'active_support'
+
+class LHS::Collection < LHS::Proxy
```
Comment (ryabrody) [inline lib/lhs/concerns/collection/internal_collection.rb]: I do not get why you are defining this class twice

Comment (pape-io) [inline lib/lhs/concerns/collection/internal_collection.rb]: It's a Concern.

**Pull request merged**